### PR TITLE
EP-2249: Connect Profile Screen to ACS Preference Center

### PR DESCRIPTION
**Pull request by reldredge71**
Add a button to link the user from the profile screen to the ACS Preference Center to update their email preferences.

### CODE REVIEW

File: src/app/profile/profile.component.js [hunk 1]
```diff
@@ -176,6 +176,12 @@ class ProfileController {
     this.pKey = this.donorDetails["acs-profile-pkey"] || this.donorDetails["acs-spouse-profile-pkey"]
   }
 
+  linkToAdobeCampaign() {
+    if (this.pKey) {
+      window.open(`https://cru-mkt-prod1-m.adobe-campaign.com/lp/LPEmailPrefCenter?_uuid=8831d67a-0d46-406b-8987-fd07c97c4ca7&_test=true&pkey=${this.pKey}&service=%400fAlW4GPmxXExp8qlx7HDlAM6FSZUd0yYRlQg6HRsO_kglfi0gs650oHPZX6LrOvg7OHoIWWpobOeGZduxdNU_m5alc`)
```
Comment (wrandall22) [inline src/app/profile/profile.component.js:181]: This URL should be environment-specific, otherwise we can't test in stage.

Comment (reldredge71) [reply src/app/profile/profile.component.js:181]: Should I just store the stage and prod URLs as constants in this file?  Or store them in `app.config.js`?  It's not just the domain names that are different, the values for `uuid` and `service` are also distinct between the two environments.

Comment (wrandall22) [reply src/app/profile/profile.component.js:181]: Probably best to put it in `app.config.js` under the `vars`. `development` can share the `staging` value.

File: src/app/profile/profile.tpl.html [hunk 2]
```diff
@@ -169,6 +169,17 @@ <h4 class="panel-title  border-bottom-small visible mt" translate>Email Addresse
...
                         </div>
                       </form>
+
+                      <div class="row">
+                        <div class="col-sm-12">
+                          <button
+                            class="btn btn-primary btn-md pull-right"
+                            ng-if="$ctrl.pKey"
+                            ng-click="$ctrl.linkToAdobeCampaign()">
+                            <translate>Email Preferences</translate>
```
Comment (wrandall22) [inline src/app/profile/profile.tpl.html:179]: There should be one button for each email.

File: src/app/profile/profile.component.js [hunk 3]
```diff
@@ -176,6 +179,12 @@ class ProfileController {
     this.pKey = this.donorDetails['acs-profile-pkey'] || this.donorDetails['acs-spouse-profile-pkey']
   }
 
+  linkToAdobeCampaign () {
+    if (this.pKey) {
+      window.open(`${this.acsUrl}?_uuid=${this.acsUuid}&_test=true&pkey=${this.pKey}&service=${this.ascServiceId}`)
```
Comment (wrandall22) [inline src/app/profile/profile.component.js:184]: Should remove the `&_test=true` here.

**Comment by wrandall22** [comment]
Looking good. All that seems to be needed now is a `git rebase master`

**Comment by wrandall22** [comment]
You can merge this into the `staging` branch locally and push up the `staging` branch. Make sure you have the latest `staging` branch before doing that though.

**Comment by reldredge71** [comment]
Okay, I pushed it to staging locally and then I pushed staging to origin/staging.  Should I close this PR?

**Comment by wrandall22** [comment]
No, don't close this PR.

**Comment by reldredge71** [comment]
@wrandall22 I made an update to this PR based on Mike's comments:
- render the Email Preferences button as a hyperlink, similar to the "Add Phone Number" button
- change the text to "Edit Email Preferences"

**Comment by reldredge71** [comment]
@wrandall22 I made some additional changes to fix the spacing.  I can squash these commits once you review them.

**Comment by wrandall22** [comment]
good to send to `staging`

**Review by wrandall22 — APPROVED** [review]

**Comment by wrandall22** [comment]
This was merged outside of the PR.

**Pull request closed without merging**